nidirect_landing_pages/nidirect_landing_pages.module

```diff
@@ -42,134 +44,29 @@ function nidirect_landing_pages_entity_update(EntityInterface $entity) {
-  // Get a list of article teasers by topic.
-  $results += nidirect_landing_pages_get_article_teasers_by_topic($tid);
-  // Alpha sort.
-  ksort($results);
-  return $results;
-}
-
-/**
- * Utility function to get teasers by term.
- */
-function nidirect_landing_pages_get_article_teasers_by_term($tid, $nid_to_exclude) {
-  // Render the 'articles by term' view and process the results.
-  $results = [];
-  $articles_view = views_embed_view('articles_by_term', 'article_teasers_by_term_embed', $tid, $tid);
-  \Drupal::service('renderer')->renderRoot($articles_view);
-  foreach ($articles_view['view_build']['#view']->result as $row) {
-    // This will be a link to an article.
-    if ($nid_to_exclude != $row->nid) {
-      $results[strtolower($row->_entity->getTitle())] = [
-        'type' => 'article_teaser',
-        'field' => 'field_article',
-        'target_id' => $row->nid,
-      ];
-    }
-  }
-  return $results;
-}
-
-/**
- * Utility function to get teasers by topic.
- */
-function nidirect_landing_pages_get_article_teasers_by_topic($tid) {
-  $results = [];
-  $articles_view = views_embed_view('site_subtopics', 'subtopic_teasers_by_topic_embed', $tid, $tid);
-  \Drupal::service('renderer')->renderRoot($articles_view);
-  foreach ($articles_view['view_build']['#view']->result as $row) {
-    // This will be a link to a taxonomy term.
-    $results[strtolower($row->_entity->getName())] = [
-      'type' => 'term_teaser',
-      'field' => 'field_term',
-      'target_id' => $row->tid,
-    ];
-  }
-  return $results;
-}
-
 /**
  * Implements hook_page_attachments().
  */
 function nidirect_landing_pages_page_attachments(array &$attachments) {
   // Add landing page js if we are creating or editing
   // a landing page node.
-  $add_js = FALSE;
-  if (\Drupal::service('router.admin_context')->isAdminRoute()) {
-    $this_route = \Drupal::routeMatch()->getRouteName();
-    if ($this_route == 'node.add') {
-      $node_type = \Drupal::routeMatch()->getParameter('node_type');
-      if ($node_type->get('type') == 'landing_page') {
-        // We are adding a landing page, add js to handle special behaviour.
-        $attachments['#attached']['library'][] = 'nidirect_landing_pages/landing_listing';
-      }
-    }
-    elseif ($this_route == 'entity.node.edit_form') {
-      $node = \Drupal::routeMatch()->getParameter('node');
-      if (is_object($node) && ($node instanceof Node) && ($node->getType() == 'landing_page')) {
-        // We are editing a landing page, add js to handle special behaviour.
-        $attachments['#attached']['library'][] = 'nidirect_landing_pages/landing_listing';
+  $this_route = \Drupal::routeMatch()->getRouteName();
+  if ($this_route == 'layout_builder.overrides.node.view') {
+    $attachments['#attached']['library'][] = 'nidirect_landing_pages/lb_config_teasers';
+    $attachments['#attached']['library'][] = 'nidirect_landing_pages/lb_config_teasers_css';
```

Comment:
Is there a reason for the 2 library entries?

Reply:
Good point, have now condensed them into one